src/components/assignment-panel/assignment-submission-info/assignment-submission-info.tsx

```diff
@@ -0,0 +1,174 @@
+                    <LegendItem label="Submitted" value={ submitted.length } color="var(--md-yellow-500)" />
+                </Tooltip>
+                <Tooltip title={ unsubmitted.map((s) => s.onyen).join(", ") }>
+                    <LegendItem label="Unsubmitted" value={ unsubmitted.length } color="var(--md-red-500)" />
+                </Tooltip>
+            </div>
+            { resubmitted.length > 0 && (
+                <div style={{ fontSize: 12, fontStyle: "italic", color: "var(--jp-ui-font-color2)" }}>
+                    *{ resubmitted.length } submission{ resubmitted.length > 1 ? "s are" : " is" } graded but out of date
+                </div>
+            ) }
+        </div>
+    )
+}
+
+export const AssignmentSubmissionInfo = ({ }: AssignmentSubmissionInfoProps) => {
+    const { assignment, students } = useAssignment()!
+
+    const [gradingActive, setGradingActive] = useState<boolean>(false)
+
+    const [graded, submitted, unsubmitted, resubmitted, total] = useMemo(() => {
+        if (!assignment || !students) return [[], [], [], [], 0]
+        let graded: IStudent[] = []
+        let submitted: IStudent[] = []
+        let unsubmitted: IStudent[] = []
+        let resubmitted: IStudent[] = []
+
+        Object.keys(assignment.studentSubmissions).forEach((onyen) => {
+            const submissions = assignment.studentSubmissions[onyen]
+            if (submissions[0]) (submissions[0] as any)._active = true
+            if (submissions[0]) (submissions[1] as any)._graded = true
+
+            const student = students.find((s) => s.onyen === onyen)!
+            // If they student has no submissions, they are unsubmitted.
+            if (submissions.length === 0) unsubmitted.push(student)
+            else {
+                const activeSubmission = submissions.find((s) => s.active)!
+                const hasGradedSubmission = submissions.some((s) => s.graded)
+
+                // The student's active submission is graded.
+                if (activeSubmission.graded) graded.push(student)
+                else {
+                    // The student has submitted but not active submission is not graded (possibly no submissions graded)
+                    submitted.push(student)
+                    // The student already has a graded submission but it isn't their active submission
+                    if (hasGradedSubmission) resubmitted.push(student)
+                }
+            }
+        })
+        return [graded, submitted, unsubmitted, resubmitted, graded.length + submitted.length + unsubmitted.length]
+    }, [assignment?.studentSubmissions, students])
+
+    const gradingDisabled = useMemo<boolean>(() => {
+        return gradingActive || (graded.length === 0 && submitted.length === 0)
+    }, [graded, submitted, gradingActive])
+
+    const grade = useCallback(async () => {
+        setGradingActive(true)
+        await new Promise((resolve) => setTimeout(resolve, 2500))
```

Comment:
This seems kinda hacky, but I probably just don't understand how we're doing it yet. Why are we just waiting 2.5s here?

Reply:
Grading isn't implemented at all yet so this was just for the purposes of debugging. Same thing applies to the setTimeout in doSync, which should be removed once the sync endpoint is actually implemented.